apps/src/lib/node/ledger/protocol/transactions/ethereum_events/mod.rs

```diff
@@ -168,9 +174,24 @@ where
             %eth_msg_keys.prefix,
             "Ethereum event already exists in storage",
         );
-        let vote_tracking =
-            calculate_updated(storage, &eth_msg_keys, voting_powers)?;
-        let changed = BTreeSet::default(); // TODO(namada#515): calculate changed keys
+        let mut votes = HashMap::default();
+        seen_by.iter().for_each(|(address, block_height)| {
+            let fvp = voting_powers
+                .get(&(address.to_owned(), block_height.to_owned()))
+                .unwrap();
+            if let Some(already_present_fvp) =
+                votes.insert(address.to_owned(), fvp.to_owned())
+            {
+                tracing::warn!(
+                    ?address,
+                    ?already_present_fvp,
+                    new_fvp = ?fvp,
+                    "Validator voted more than once, arbitrarily using later value",
+                )
+            }
```

Comment:
change `fvp` to `fractional_voting_power`, or even just `voting_power`. the former is somewhat confusing